src/Sentry/Internal/MainExceptionProcessor.cs

```diff
@@ -110,14 +110,24 @@ internal static Mechanism GetMechanism(Exception exception)
         {
             Debug.Assert(exception != null);
 
-            Mechanism mechanism = null;
+            Mechanism mechanism = new Mechanism()
+            {
+                Handled = true
+            };
 
             if (exception.HelpLink != null)
             {
-                mechanism = new Mechanism
-                {
-                    HelpLink = exception.HelpLink
-                };
+                mechanism.HelpLink = exception.HelpLink;
+            }
+            if (exception.Data?.Contains("Sentry:Handled") == true)
+            {
+                mechanism.Handled = (exception.Data["Sentry:Handled"] as bool?).Value;
+                exception.Data.Remove("Sentry:Handled");
+            }
+            if (exception.Data?.Contains("Sentry:Mechanism") == true)
+            {
+                mechanism.Type = exception.Data["Sentry:Mechanism"] as string;
+                exception.Data.Remove("Sentry:Mechanism");
```

Comment:
Now we have two lookups in the hashtable:

```
            if (exception.Data?.Contains("Sentry:Handled") == true)
```

And then a second one:

```
                 mechanism.Handled = (exception.Data["Sentry:Handled"] as bool?).Value;
```

Ideally we'd use `TryGetValue` but since this is the old school `IDictionary`, it's not available.
We could at least use the indexers and pattern matching:

```suggestion
        // Throw these at the top of the Mechanism class so it can be used by AppDomainUnhandledExceptionIntegration.cs too
        // Can be used from the unit tests too
        public static readonly string HandledKey = "Sentry:Handled";
        public static readonly string MechanismKey = "Sentry:Mechanism";
        
            if (exception.Data[HandledKey] is bool handled)
            {
                mechanism.Handled = handled;
                exception.Data.Remove(HandledKey);
            }
            if (exception.Data[MechanismKey] is string mechanismName)
            {
                mechanism.Type = mechanismName;
                exception.Data.Remove(MechanismKey);
```